### ISSUE TITLE
Error in append literal of VirtuosoRepositoryConnection

### ISSUE THREAD

**Issue by slater2022** [openlink/virtuoso-opensource#1056]
```
private void append(Literal lit, StringBuilder sb, boolean inTriplePattern)
            throws RepositoryException
    {
        sb.append("\"");
        escapeString(lit.getLabel(), sb);
        sb.append("\"");

        if (lit.getDatatype() != null) {
            // Append the literal's datatype
            IRI ltype = lit.getDatatype();
            if (!(insertStringLiteralAsSimple && ltype.equals(XMLSchema.STRING))) {
                sb.append("^^");
                append(ltype, sb, inTriplePattern);
            }
        }
        else if (lit.getLanguage() != null) {
            // Append the literal's language
            sb.append("@");
            sb.append(lit.getLanguage());
        }
    }
```
When `lit` has a language tag, `lit.getDatatype()` is `"http://www.w3.org/1999/02/22-rdf-syntax-ns#langString"`, will not execute `"else if (lit.getLanguage() != null)"`

**Comment by HughWilliams** [comment]
We are looking into this ...

**Comment by imitko** [comment]
@slater2022 

I don't think you can have both language tag and datatype IRI ref, as per [RDFLiteral](https://www.w3.org/TR/rdf-sparql-query/#rRDFLiteral) rule. Thus the if switch have to be transposed.

HTH

**Comment by pkleef** [comment]
This has been fixed in df1cf9f0e22afe4625243f71227557039df60980 on the `develop/7` branch.

**Comment by slater2022** [comment]
Okay, it's working
